frontend/src/features/settings/preferences/GenericPollUserSettingsForm.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { Button, Grid, TextField, Typography } from '@mui/material';
+
+import {
+  replaceSettings,
+  selectSettings,
+} from 'src/features/settings/userSettingsSlice';
+import { useNotifications } from 'src/hooks';
+import {
+  ApiError,
+  TournesolUserSettings,
+  UsersService,
+} from 'src/services/openapi';
+
+interface Props {
+  // Allowed polls. Using literal allows us to use the form settings[pollName]
+  // with TypeScript.
+  pollName: 'videos';
```

Comment:
Let's see if we can avoid to hardcode the polls' names here.

Reply:
We could use `keyof` but the resulting type would include not only the polls but also all other additional keys that the object `TournesolUserSettings` would contain, such as the future global settings. 

```javascript
pollName: keyof TournesolUserSettings;
```

Reply:
We could use the directly the `polls` array defined in the file `constants.tsx` to list all polls, but it doesn't work.

This array contains the poll `presidentielles2022` which is not a key of `TournesolUserSettings`. In fact, the prop `pollName` should not allow all polls, but only those defined by `TournesolUserSettings` to avoid triggering TypeScript errors.

```javascript
// Let's pretend 'videos' and 'presidentielle' come from polls.map(poll => poll.name);
const pollNames = ['videos', 'presidentielle2022'] as const;

interface Props {
  pollName: typeof pollNames[number];
}

// ...

// The following line produce an error as `pollName` can be 'presidentielle2022', which is not
// a key of `userSettings`.
const [rateLaterAutoRemove, setRateLaterAutoRemove] = useState(
  userSettings ? userSettings[pollName]?.rate_later__auto_remove ?? 4 : 4
);
```
```

Reply:
I finally opted for the form:

```javascript
interface Props {
  pollName: typeof YOUTUBE_POLL_NAME;
}
```

It could be better if it was possible to extract the allowed polls from the constant `polls`, but I don't know if it's possible with TypeScript. My idea was:

```javascript
// This line produces the error:
//   A 'const' assertions can only be applied to references to enum members,
//   or string, number, boolean, array, or // object literals.
const pollNames = polls.filter(poll => poll.hasSettings === true).map(poll => poll.name) as const;

interface Props {
  pollName: typeof pollNames[number];
}
```